scipy/signal/tests/test_signaltools.py

```diff
@@ -3741,24 +3740,26 @@ def test_single_1dperiod(self, xp):
 
         assert strength.ndim == 0
         assert phase.ndim == 0
-        assert_almost_equal(strength, targ_strength)
-        assert_almost_equal(phase, 2 * np.pi * targ_phase)
 
+        assert_almost_equal(xp.asarray(strength), xp.asarray(targ_strength))
+        assert_almost_equal(xp.asarray(phase), xp.asarray(2 * xp.pi * targ_phase))
```

Comment:
why do we need to call `xp.asarray` on `strength` and `phase`? Are we not always returning a 0-D array?

same below

Reply:
```
In [5]: xp_assert_close(1.0, 2)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[5], line 1
----> 1 xp_assert_close(1.0, 2)

    [... skipping hidden 1 frame]

File ~/.conda/envs/scipy-dev/lib/python3.11/site-packages/scipy/_lib/_array_api.py:245, in _strict_check(actual, desired, xp, check_namespace, check_dtype, check_shape)
    242 desired = xp.asarray(desired)
    244 if check_dtype:
--> 245     _msg = f"dtypes do not match.\nActual: {actual.dtype}\nDesired: {desired.dtype}"
    246     assert actual.dtype == desired.dtype, _msg
    248 if check_shape:

AttributeError: 'float' object has no attribute 'dtype'
```

And no, this beautiful little functions is not always returning 0D arrays.